internal/bgp/bgp.go

```diff
@@ -665,3 +679,30 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the provided address exists locally on
+// any of the host's network interfaces. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(addr string) bool {
+	interfaces, err := net.Interfaces()
```

Comment:
I'd add a check for:
```
if addr == nil {
return false
}
```

So this is clearly a no-op when the string wasn't set and after that you can log. If you are doing something after, it is worth logging as *some* address was specified and the user expects a behavior. So, doing that will make logging just easy too :-)